main/java/com/bchupika/user/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.bchupika.user.repository;
+
+import com.bchupika.user.dto.UserRegistrationDto;
+import com.bchupika.user.entity.User;
+import com.bchupika.user.exception.DataProcessingException;
+import com.bchupika.utils.ConnectionUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Optional;
+import java.sql.SQLException;
+
+public class UserRepositoryImpl implements UserRepository {
+    public Number save(UserRegistrationDto user) throws DataProcessingException {
+        String sqlRequest = "INSERT INTO user (name, last_name, email, phone, password) VALUES (?, ?, ?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sqlRequest,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, user.getName());
+            preparedStatement.setString(2, user.getLastName());
+            preparedStatement.setString(3, user.getEmail());
+            preparedStatement.setString(4, user.getPhoneNumber());
+            preparedStatement.setString(5, user.getPhoneNumber());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                return generatedKeys.getObject(1, Long.class);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create instance of user:"
+                    + user.toString()
+                    + " and add it to DB " + e.getMessage(), e);
+        }
+        return 0;
```

Comment:
not sure we should return 0 in this case. If we don't have result here `if (generatedKeys.next())` maybe it is better to throw an exception?

And it is better to return `User` instead of `Number`